examples/federated/core_package/index.js

```diff
@@ -48,29 +53,32 @@ async function main() {
   const extension_data = JSON.parse(
     PageConfig.getOption('dynamic_extensions')
   );
-  const mime_extension_data = JSON.parse(
-    PageConfig.getOption('dynamic_mime_extensions')
-  );
+
+  const dynamicPlugins = [];
+  const dynamicMimePlugins = [];
 
   // Get dynamic plugins
   // TODO: deconflict these with builtins?
-  const dynamicPromises = extension_data.map(data =>
-    loadComponent(
-      data.path,
-      data.name,
-      data.module
-    )
-  );
-  const dynamicPlugins = await Promise.all(dynamicPromises);
-
-  const dynamicMimePromises = mime_extension_data.map(data =>
+  const componentLoadPromises = extension_data.map((data) =>
     loadComponent(
-      data.path,
-      data.name,
-      data.module
+      `${URLExt.join(PageConfig.getOption('fullLabextensionsUrl'), data.name, 'remoteEntry.js')}`,
+      data.name
     )
+    .then(() => Promise.all([
+      createModule(data.name, data.plugin),
+      createModule(data.name, data.mimePlugin),
+      createModule(data.name, data.style)
```

Comment:
Thanks, I think this is better. I was thinking of something more like:

```typescript
let c = loadComponent(...)
if (data.plugin) {
  dynamicPlugins.push(c.then(() => createModule(data.name, data.plugin)));
}
if (data.mimePlugin) {
  dynamicMimePlugins.push(c.then(() => createModule(data.name, data.mimePlugin)));
}
if (data.style) {
  c.then(() => createModule(data.name, data.style));
}
```
Then each module *only* depends on the script file being loaded, rather than having unnecessary synchronization barriers, where the dynamic plugin depends on the mime plugin being loaded first.

Reply:
Yeah, make sense.
I revised the code but in order to do it this way I had to add some safety wrappers.
Thank you,